Small fixes and changes to GL, VK, D3D11 and D3D12 pixel history tests

Zorro666 opened the pull request:
## Description

- Vk & D3D12: Check the post mod stencil at pixel location 110,100
- Vk & D3D12: Check the post mod stencil is unknown at pixel location 275, 260 for fragment zero of a multi-fragment event
- Vk, D3D11 & D3D12: Validate that in a fragment event the postMod stencil value should be unknown (-1 or -2)
- This test validates the D3D12 pixel history code change in 4815adaa83b
- Tidy up python logging in GL Pixel History and marked it as a slow test
- D3D12 Pixel History set fragment postMod stencil to unknown (-2)

Zorro666 committed:
Extend Vk Pixel History tests

Check the post mod stencil at pixel location 110,100
Check the post mod stencil is unknown at pixel location 275, 260 for fragment zero of a multi-fragment event

Validate that in a fragment event the postMod stencil value should be unknown (-1 or -2)

Zorro666 committed:
Extend D3D12 Pixel History tests

Check the post mod stencil at pixel location 110,100
Check the post mod stencil is unknown at pixel location 275, 260 for fragment zero of a multi-fragment event

Validate that in a fragment event the postMod stencil value should be unknown (-1 or -2)
This test validates the code change in 4815adaa83b

skarolewics commented:
I don't think that -1 or -2 is a valid stencil value in D3D12, as it's a UINT8 format. Is there a Vulkan extension or API that permits a signed stencil format?

Zorro666 committed:
Tidyup python logging in GL Pixel History

Zorro666 committed:
Extend D3D11 Pixel History tests

Validate that in a fragment event the postMod stencil value should be unknown (-1 or -2)

Zorro666 committed:
D3D12 Pixel History set fragment postMod stencil to unknown (-2)

Zorro666 commented:
> I don't think that -1 or -2 is a valid stencil value in D3D12, as it's a UINT8 format. Is there a Vulkan extension or API that permits a signed stencil format?

-1 and -2 are RenderDoc sentinel values for the stencil value to mark the stencil as unknown. It is used by the UI to display the "-" (-1) and "?" (-2) for the stencil value. The -1 or -2 can't come from an actual stencil value generated by a drawcall. The -1 or -2 is set by the RenderDoc pixel history replay code.

Zorro666 committed:
Marked GL Pixel History as a slow test

skarolewics commented:
Gotcha, thanks for the info. I had no idea that RenderDoc did that, so it's not surprising that the D3D12 code doesn't handle it correctly.

Zorro666 commented:
Thanks Steve - it was a single line change to fix the unknown stencil setting for D3D12. 
D3D12 Pixel history is shaping up nicely thanks to your hard work to get the foundation and base implementation in place. It is really helpful to have the test harness available for D3D12 pixel history. 
There is also a PR to fix the failing MSAA test #3193
